pkgs/development/interpreters/python/fetch-python-requirements.nix

```diff
@@ -0,0 +1,181 @@
+    pipFlags ? [],
+
+    # Throw an error by default to warn the user about the correct usage
+    targetSystem ? throw ''
+      'targetSystem' must be specified for fetchPythonRequirements.
+      Pass only a constant string like for example "x86_64-linux".
+      Do not pass `stdenv.targetPlatform.system` or `builtins.currentSystem`.
+      This prevents the fetcher from being executed on the wrong platform.
+    '',
+  }:
+
+    # specifying `--platform` for pip download is only allowed in combination with `--only-binary :all:`
+    # therefore, if onlyBinary is disabled, we must enforce targetPlatform == buildPlatform to ensure reproducibility
+    if ! onlyBinary && targetSystem != stdenv.buildPlatform.system then
+      throw ''
+        fetchPythonRequirements cannot fetch sdist packages for ${targetSystem} on a ${stdenv.buildPlatform.system}.
+        Either build on a ${targetSystem} or set `onlyBinary = true`.
+      ''
+    else
+    let
+      # map nixos system strings to python platforms
+      sysToPlatforms = {
+        "x86_64-linux" = [
+          "manylinux1_x86_64"
+          "manylinux2010_x86_64"
+          "manylinux2014_x86_64"
+        ];
+        "x86_64-darwin" =
+          lib.forEach (lib.range 0 15) (minor: "macosx_10_${builtins.toString minor}_x86_64");
+        "aarch64-linux" = [
+          "manylinux1_aarch64"
+          "manylinux2010_aarch64"
+          "manylinux2014_aarch64"
+        ];
+      };
+      platforms = if sysToPlatforms ? "${targetSystem}" then sysToPlatforms."${targetSystem}" else throw ''
+        'binaryOnly' fetching is currently not supported for target ${targetSystem}.
+        You could set 'binaryOnly = false' and execute the build on a ${targetSystem}.
+      '';
+
+      pythonAttrName = lib.stringAsChars (x: if x == "." then "" else x) python.libPrefix;
+
+      # We need to execute pip from the python version for which we want to download packages.
+      # Pip accepts '--python-version', but this works only for wheel packages.
+      pythonPip = buildPackages."${pythonAttrName}".withPackages (ps: [
+        ps.pip
+        ps.setuptools
+        ps.pkgconfig
+      ]);
+
+      # Normalize package names:
+      #   "Example_Package[extra]" -> "example-package"
+      normalizedNames = map (req:
+        lib.toLower (lib.replaceStrings [ "_" ] [ "-" ] (lib.elemAt (lib.splitString "[" req) 0))
+      ) requirements;
+
+      # fixed output derivation containing packages,
+      # each being symlinked from it's normalized name
+      # Example:
+      #   "$out/werkzeug" will point to "$out/Werkzeug-0.14.1-py2.py3-none-any.whl"
+      self = stdenv.mkDerivation {
+
+        name = "python-sources-${targetSystem}";
+
+        outputHashMode = "recursive";
+        outputHashAlgo = "sha256";
+        inherit outputHash;
+
+        nativeBuildInputs = with python3.pkgs; [ pythonWithMitmproxy pythonPip curl cacert ];
```

Comment:
```suggestion
        nativeBuildInputs = with python3.pkgs; [ pythonWithMitmproxy pythonPip ] ++ [ curl cacert ];
```

Reply:
Nothing in that list is from `python3.pkgs`, so I removed with `with` statement now.